GLAA.Web/FormLogic/LicenceApplicationFormDefinition.cs

```diff
@@ -38,11 +52,115 @@ public bool CanViewNextModel<TParent>(FormSection section, int id, TParent paren
             return model == null || model.CanView(parent);
         }
 
+        public bool CanViewPage<TParent>(FormSection section, string actionName, TParent parent)
+        {
+            var page = GetPageDefinition(section, actionName);
+
+            if (page.OverrideViewCondition)
+            {
+                return true;
+            }
+
+            var model = page.GetViewModelExpressionForPage(parent) as ICanView<TParent>;
+
+            return model == null || model.CanView(parent);
+        }
+
         public int GetSectionLength(FormSection section)
         {
-            return fieldConfiguration.Fields[section].Length;
+            return fieldConfiguration[section].Length;
         }
 
+        public FormPageDefinition GetLastPage(FormSection section)
+        {
+            return fieldConfiguration[section].Last();
+        }
+
+        public bool IsNextPageLastPage(FormSection section, string actionName)
```

Comment:
huh? NextPageLastPage? Sounds a little confusing!

Reply:
Yeah, I'm going to go through and rename everything once I'm done changing things. There's a few bits like this that make my brain hurt!